--- a/src/vtsspp/Makefile
+++ b/src/vtsspp/Makefile
@@ -106,7 +106,7 @@ $(DRIVER_NAME)-objs: COPY_PUBLIC_HEADERS
 _KBUILD_CFLAGS := $(call flags,KBUILD_CFLAGS)
 CHECK_BUILD    := $(CC) $(KBUILD_CPPFLAGS) $(CPPFLAGS) $(LINUXINCLUDE) \
                   $(_KBUILD_CFLAGS) $(CFLAGS_KERNEL) $(EXTRA_CFLAGS) $(CFLAGS) \
-                  -DKBUILD_BASENAME=\"$(DRIVER_NAME)\" -Werror -S -o /dev/null -xc >> $(obj)/.autoconf.log 2>&1
+                  -DKBUILD_BASENAME=\"$(DRIVER_NAME)\" -Werror -S -Wno-pointer-sign -o /dev/null -xc >> $(obj)/.autoconf.log 2>&1
 
 autoconf-srcs := $(sort $(wildcard $(M)/autoconf/*.c))
 $(addprefix $(obj)/,$($(DRIVER_NAME)-objs)): $(obj)/autoconf.h
